include/onnx-mlir/Runtime/OnnxDataType.h

```diff
@@ -43,6 +43,18 @@ extern const char *OM_DATA_TYPE_NAME[];
 #ifdef __cplusplus
 // Note by design const map has no [] operator since [] creates a default
 // key value mapping when the key is not found which changes the map
+#if defined(_WIN32)
+// clang-format off
+const std::map<std::string, OM_DATA_TYPE> OM_DATA_TYPE_CPP_TO_ONNX = {
```

Comment:
It's because of `typeid(T).name` returns different value on Windows. We use `typeid` here `omt->_dataType = OM_DATA_TYPE_CPP_TO_ONNX.at(std::string(typeid(T).name()));` in https://github.com/onnx/onnx-mlir/blob/main/src/Runtime/OMTensor.inc#L535